llvm/lib/IR/AutoUpgrade.cpp

```diff
@@ -5461,6 +5461,13 @@ std::string llvm::UpgradeDataLayoutString(StringRef DL, StringRef TT) {
     return Res;
   }
 
+  if (T.isSPARC()) {
+    // Add "-i128:128"
+    if (!DL.empty() && !DL.contains("-i128:128"))
+      Res.append("-i128:128");
+    return Res;
+  }
```

Comment:
Could you please also add a test for this in llvm/unittests/Bitcode/DataLayoutUpgradeTest.cpp?

I also believe that this needs to be inserted at the correct position, not just appended, so that the string would exactly match the new one. I think there will be assertion failures somewhere otherwise, but I'm not entirely sure.